src/SourceBuild/content/eng/tools/tasks/Microsoft.DotNet.SourceBuild.Tasks.XPlat/UpdateNuGetConfigPackageSourcesMappings.cs

```diff
@@ -102,75 +93,97 @@ public override bool Execute()
             XElement pkgSrcMappingClearElement = pkgSrcMappingElement.Descendants().FirstOrDefault(e => e.Name == "clear");
 
             // When building online add the filtered mappings from original online sources.
+            // If there are none, add default mappings for all online sources.
             if (BuildWithOnlineFeeds)
             {
-                foreach (DictionaryEntry entry in oldSourceMappingPatterns)
+                if (oldSourceMappingPatterns.Count > 0)
                 {
-                    // Skip sources with zero package patterns
-                    if (entry.Value == null)
+                    foreach (var entry in oldSourceMappingPatterns)
                     {
-                        continue;
+                        // Skip sources with zero package patterns
+                        if (entry.Value != null)
+                        {
+                            pkgSrcMappingClearElement.AddAfterSelf(GetPackageMappingsElementForSource(entry.Key, entry.Value));
+                        }
                     }
-
-                    XElement pkgSrc = new XElement("packageSource", new XAttribute("key", entry.Key));
-                    foreach (string pattern in (List<string>)entry.Value)
+                }
+                else
+                {
+                    foreach (XElement sourceElement in pkgSourcesElement.Descendants().Where(e => e.Name == "add" && !allSourcesPackages.Keys.Contains(e.Attribute("key").Value)))
                     {
-                        pkgSrc.Add(new XElement("package", new XAttribute("pattern", pattern)));
+                        pkgSrcMappingClearElement.AddAfterSelf(new XElement("packageSource", new XAttribute("key", sourceElement.Attribute("key").Value), new XElement("package", new XAttribute("pattern", "*"))));
                     }
-
-                    pkgSrcMappingClearElement.AddAfterSelf(pkgSrc);
                 }
             }
 
             // Add package source mappings for local package sources
             foreach (string packageSource in allSourcesPackages.Keys)
             {
                 // Skip sources with zero package patterns
-                if (allSourcesPackages[packageSource] == null)
+                if (allSourcesPackages[packageSource] != null)
                 {
-                    continue;
+                    pkgSrcMappingClearElement.AddAfterSelf(GetPackageMappingsElementForSource(packageSource, allSourcesPackages, currentPackages, previouslyBuiltPackages));
                 }
+            }
+
+            using (var writer = XmlWriter.Create(NuGetConfigFile, new XmlWriterSettings { NewLineChars = newLineChars, Indent = true }))
+            {
+                document.Save(writer);
+            }
+
+            return true;
+        }
+
+        private void AddDefaultMappingsForOnlineSources(XElement pkgSrcMappingClearElement, XElement pkgSourcesElement)
+        {
+            throw new NotImplementedException();
```

Comment:
Is this intentional or an incomplete refactoring?

Reply:
Not intentional - not used as I refactored it more. Will fix.

Reply:
Fixed with https://github.com/dotnet/installer/pull/19114/commits/3f458e5268b5925b8fae42f35138b3f344b24f28